test/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildALibrary.cs

```diff
@@ -66,6 +68,109 @@ public void It_builds_the_library_twice_in_a_row()
                 .Pass();
         }
 
+        [Fact]
+        public void It_ignores_excluded_folders()
+        {
+            Action<GetValuesCommand> setup = getValuesCommand =>
+            {
+                foreach (string folder in new[] { "bin", "obj", "packages" })
+                {
+                    WriteFile(Path.Combine(getValuesCommand.ProjectRootPath, folder, "source.cs"),
+                        "!InvalidCSharp!");
+                }
+
+                WriteFile(Path.Combine(getValuesCommand.ProjectRootPath, "Code", "Class1.cs"),
+                    "public class Class1 {}");
+            };
+
+            var compileItems = GetItemsFromTestLibrary("Compile", setup);
+
+            compileItems = compileItems.Where(i =>
+                    !i.EndsWith("AssemblyAttributes.cs", System.StringComparison.OrdinalIgnoreCase) &&
+                    !i.EndsWith("AssemblyInfo.cs"))
```

Comment:
One is for the `TargetFrameworkAttribute` and comes from the temp folder.  The other is for the other assembly attributes that the SDK adds, and is in the intermediate output path.

I've factored this out into a separate method that removes these, with a comment to explain what's going on.